torchvision/transforms/functional.py

```diff
@@ -161,8 +161,8 @@ def convert_image_dtype(image: torch.Tensor, dtype: torch.dtype = torch.float) -
             msg = f"The cast from {image.dtype} to {dtype} cannot be performed safely."
             raise RuntimeError(msg)
 
-        eps = 1e-3
-        return image.mul(torch.iinfo(dtype).max + 1 - eps).to(dtype)
```

Comment:
this was giving results that didn't match PIL

Reply:
Hum, we had quite a lot of discussion about this behavior in https://github.com/pytorch/vision/pull/2078#issuecomment-612045321. I believe if we make the multiplication go to `dtype.max`, we will end up with a non-uniform distribution over the last values.

@pmeier thoughts?

Reply:
@fmassa is right about the intention. I think this boils down to 
- do we want it "right" or
- do we want it compatible to other packages.

I'm in favor for the former (hence my implementation), but I can see why the latter is also feasible.

Reply:
I think my issue was internal consistency, not a difference between us and PIL. That comment thread is helpful -- we certainly expect more than one floating point value to map to 255.

I was able to fix it by just making the gamma adjustment consistent with `convert_image_dtype`.

This raises another issue though -- I'm not sure if it's still OK to express the equation for adjust_gamma as `255 * gain * (img/255) ** gamma` in the docs, where in reality it's `255.999 * gain....`. I want to be accurate but also don't want to be unnecessarily confusing, and the doc does say "based on."